components/RecipeDetails.js

```diff
@@ -1,35 +1,54 @@
 import styled from "styled-components";
 import Image from "next/image";
 import Link from "next/link";
+import TrashBin from "public/svg/TrashBin.svg";
+import { useRouter } from "next/router";
+import { v4 as uuidv4, v4 } from "uuid";
+
+export default function RecipeDetails({ recipeDetails, onDeleteRecipe }) {
+  const router = useRouter();
+
+  function handleDeleteAndReroute() {
+    if (confirm("Are you sure you want to delete this recipe?")) {
+      onDeleteRecipe(recipeDetails.id);
+      router.push("/");
+    }
+  }
 
-export default function RecipeDetails({ recipeDetails }) {
   return (
-    <RecipeCard>
-      <h1> {recipeDetails.name}</h1>
-      <StyledImage
-        src={
-          recipeDetails.imageURL
-            ? recipeDetails.imageURL
-            : "/recipe-images/generic-food-3.png"
-        }
-        alt={recipeDetails.name}
-        width={300}
-        height={300}
-      />
-      <p>⏰ {recipeDetails.preparationTime}</p>
-      <ul>
-        {recipeDetails.ingredients.map(
-          (ingredient) =>
-            ingredient.ingredient && (
-              <li key={ingredient.ingredient}>
-                {ingredient.quantity} - {ingredient.ingredient}
-              </li>
-            )
-        )}
-      </ul>
-      <StyledDescription>{recipeDetails.description}</StyledDescription>
-      <StyledLink href="/">⬅ Back</StyledLink>
-    </RecipeCard>
+    <>
+      <RecipeCard>
+        <h1> {recipeDetails.name}</h1>
+        <StyledImage
+          src={
+            recipeDetails.imageURL
+              ? recipeDetails.imageURL
+              : "/recipe-images/generic-food-3.png"
+          }
+          alt={recipeDetails.name}
+          width={300}
+          height={300}
+        />
+        <p>⏰ {recipeDetails.preparationTime}</p>
+        <ul>
+          {recipeDetails.ingredients.map(
+            (ingredient) =>
+              ingredient.name && (
+                <li key={uuidv4()}>
```

Comment:
to be safe you should change your datastructure once more and give your ingredients an ``id``. And then use it here.

Reply:
We have over 80 ingredients inside the dummy data. Should we manually add a id for every entry?

I want to note that we completely rework the ingredients at the user story [Add and delete ingredients#16](https://github.com/feivlinus/fork-yeah/issues/16)